src/Workspaces/Core/Portable/Workspace/Host/TextFactory/TextFactoryService.cs

```diff
@@ -17,5 +17,12 @@ public SourceText CreateText(Stream stream, Encoding defaultEncoding, Cancellati
             cancellationToken.ThrowIfCancellationRequested();
             return SourceText.From(stream, defaultEncoding);
         }
+
+        public SourceText CreateText(TextReader reader, Encoding encoding, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
The text reader could represent a large file that get's read in a loop.  We need ability to quickly abort work.